include/extractor/scripting_environment.hpp

```diff
@@ -65,8 +77,14 @@ class ScriptingEnvironment
                     tbb::concurrent_vector<std::pair<std::size_t, ExtractionWay>> &resulting_ways,
                     tbb::concurrent_vector<boost::optional<InputRestrictionContainer>>
                         &resulting_restrictions) = 0;
+
+    static location_handler_sparse_type *location_sparse_handler;
+    static location_handler_dense_type *location_dense_handler;
+    static void setSparseCache(location_handler_sparse_type &cache) {
+        location_sparse_handler = &cache;};
+    static void setDenseCache(location_handler_dense_type &cache) {
+        location_dense_handler = &cache;};
```

Comment:
Ok, i think i understand.
instead of create a libosmium cache I should use the 
`stxxl::vector<ExternalMemoryNode>;`
who store data into the extraction_containers.
And split the `for` loop reader into two loop, to parse in first time Node and after that other geometry.